### DISCUSSION TITLE
Run script at boot

### DISCUSSION THREAD

**Discussion by drderiv** [tesla-android/issue-tracker#279] [Q&A]
I was able to gain i2c control over the fan in my Argon case.  However, right now I have to adb in after each boot and do an insmod.  How can I get the driver to load at each boot?  Best would be if I could run a script at boot, then I could load the driver and set up temperature-based control.  Is there a user init folder where I can add something like that?

**Comment by mikegapinski** [comment]
I can just bundle your files into the system; no problem. Can you send me everything you did to control the fan? 

You can add the script to /vendor/init/*.rc

One thing; if you did adb remount you might not be able to OTA to newer versions

**Comment by cjncodes** [comment]
Perhaps this is similar to what I’m looking to do. Using the Argon Fan Hat, they say to open up Terminal and **curl https://download.argon40.com/argon fan hat.sh | bash**. I’m admittedly new to Android and just got my hands on a Pi 4 to start using Tesla Android (thank you!!). Happy to test/try whatever’s needed!
